src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -1474,5 +1508,112 @@ public void OnBeforeQueryInvoke(BeforeQueryEventArgs beforeQueryEventArgs)
 
             return (indexName, collectionName);
         }
+
+        /// <summary>
+        /// an internal structure that is used for caching counter values in a session
+        /// </summary>
+        protected internal class CountersCache
+        {
+            public CountersCache()
+            {
+                Values = new Dictionary<string, long?>(StringComparer.OrdinalIgnoreCase);
+            }
+
+            public Dictionary<string, long?> Values { get; set; }
+
+            public bool GotAll
+            {
+                get => _gotAll;
+                set
+                {
+                    _gotAll = value;
+
+                    if (value)
+                        _missingCounters = null;
+                }
+            }
+
+            private bool _gotAll { get; set; }
+            private HashSet<string> _missingCounters;
```

Comment:
What is the purpose of this?
Why not just put `Values[counter] = null` ?

Reply:
removed _missingCounters